plumbing/object/commitgraph/commitnode_graph_builder.go

```diff
@@ -0,0 +1,231 @@
+package commitgraph
+
+import (
+	"errors"
+	"math"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	commitgraph "github.com/go-git/go-git/v5/plumbing/format/commitgraph/v2"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/storer"
+)
+
+type CreateCommitNodeGraphOptions struct {
+	Commits object.CommitIter // If nil then all commits in the storage are used as potential heads, otherwise only the commits in the iter are used
+	Append  bool              // Append all commits in the old index to the new index
+	Chain   bool              // Chain the new index to the old index (only valid if Append is false and the old index has generation v2)
```

Comment:
```suggestion
	// Commits holds the commits to be iterated upon. If nil, then all commits in the storage are used as potential heads, otherwise only the commits in the iter are used.
	Commits object.CommitIter
	// Append defines whether all commits in the old index should be appended to the new index.
	Append  bool
        // Chain defines whether the new index should be chained to the old index. Can only be used if Append is false and the old index has generation v2.
	Chain   bool
```